### PR TITLE
Repo-1102: Redlands validations

### PR DESCRIPTION
**Pull request by okeen**
Refactors Bulkrax import and export validations to be more generic.
Adds Csv Validator.

### CODE REVIEW

File: app/services/hyku_addons/validations/entry_validation_service.rb [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        @entry.current_status.update(error_backtrace: @errors)
+        false
+      end
+
+      def source_metadata
+        @_source_metadata ||= @entry.raw_metadata
+      end
+
+      def destination_metadata
+        @_destination_metadata ||= ActiveFedora::SolrService.get("id:#{@entry.identifier}").dig('response', 'docs')&.first || {}
```
Comment (prdanelli) [inline app/services/hyku_addons/validations/entry_validation_service.rb:34]: the ' here.

File: app/services/hyku_addons/validations/solr_entry_validation_service.rb [hunk 2]
```diff
@@ -0,0 +1,183 @@
+# frozen_string_literal: true
+
+module HykuAddons
+  module Validations
+    class SolrEntryValidationService < EntryValidationService
+      attr_reader :errors, :entry
+
+      SOURCE_SERVICE_OPTIONS = {
+        base_url: ENV['BULKRAX_SOURCE_BASE_URL'],
```
Comment (prdanelli) [inline app/services/hyku_addons/validations/solr_entry_validation_service.rb:9]: the ' here.

File: app/services/hyku_addons/validations/solr_entry_validation_service.rb [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        def reevaluate_has_model_ssim(old_value)
+          ["Pacific#{gross_work_type_name(old_value)}"]
+        end
+
+        def reevaluate_human_readable_type_tesim(old_value)
+          ["Pacific #{gross_work_type_name(old_value)}"]
+        end
+
+        def reevaluate_admin_set_tesim(old_value)
+          Array.wrap(old_value).first == "Default Admin Set" ? ['Default'] : old_value
```
Comment (prdanelli) [inline app/services/hyku_addons/validations/solr_entry_validation_service.rb:123]: the mix of ' and " here.

**Review by prdanelli — DISMISSED** [review]

File: app/models/hyku_addons/csv_entry.rb [hunk 4]
```diff
@@ -167,7 +167,7 @@ def build_json_metadata
     def export_mapping
       # Add all fields from this model to the mapping unless explicitly excluded by already being in the mapping
       map = mapping.reject { |k, _| k == 'model' || (Bulkrax.reserved_properties.include?(k) && !field_supported?(k)) }
-      hyrax_record.attributes.keys.map { |k| map[k] = { 'from' => [k] } unless map.key?(k) || Bulkrax.reserved_properties.include?(k) || !field_supported?(k) }
+      hyrax_record.attributes.keys.map { |k| map[k] ||= { 'from' => Array.wrap(k) } unless Bulkrax.reserved_properties.include?(k) || !field_supported?(k) }
```
Comment (cjcolvar) [inline app/models/hyku_addons/csv_entry.rb:170]: :+1:

File: app/services/hyku_addons/validations/solr_entry_validation_service.rb [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        password: ENV['BULKRAX_SOURCE_PASSWORD']
+      }.with_indifferent_access.freeze
+
+      DESTINATION_SERVICE_OPTIONS = {
+        base_url: ENV['BULKRAX_DESTINATION_BASE_URL'],
+        username: ENV['BULKRAX_DESTINATION_USERNAME'],
+        password: ENV['BULKRAX_DESTINATION_PASSWORD']
+      }.with_indifferent_access.freeze
+
+      EXCLUDED_FIELDS = %i[
```
Comment (cjcolvar) [inline app/services/hyku_addons/validations/solr_entry_validation_service.rb:20]: A lot of the values and logic in this file were probably specific to the hyku1 migration we did so maybe it makes sense to rename this or slowly refine those idiosyncrasies away over time (or into a specific subclass if we want to keep them around).  Nothing needs to be done for this PR but just thoughts for later.

**Review by cjcolvar — APPROVED** [review]
Looks good to me.  I think `'` vs. `"` will have to come with a rubocop cop so I expect that to be in its own PR with just those changes throughout the codebase.

**Review by cjcolvar — APPROVED** [review]

**Pull request merged**